src/main/scala/com/fulcrumgenomics/umi/DuplexConsensusCaller.scala

```diff
@@ -139,6 +139,14 @@ class DuplexConsensusCaller(override val readNamePrefix: String,
         val (_, abR1s, abR2s) = subGroupRecords(ab)
         val (_, baR1s, baR2s) = subGroupRecords(ba)
 
+        // The orientation of AB and BA reads should be:
+        // AB R1: +  AB R2: -
+        // BA R1: -  BA R2: +
+        // Therefore, AB-R1s and BA-R2s should be on the same strand, and the same for AB-R2s and BA-R1s
+        // Check for this explicitly here.
+        require((abR1s ++ baR2s).map(_.getReadNegativeStrandFlag).distinct.size <= 1, "Not all AB-R1s and BA-R2s were on the same strand")
```

Comment:
Also would be nice to print out something here that would allow tracking back to the BAM records when there is a failure.  E.g. including the `MI` in the `require()` message.